lib/msal-common/src/authority/AuthorityFactory.ts

```diff
@@ -36,10 +36,12 @@ export class AuthorityFactory {
         correlationId?: string
     ): Promise<Authority> {
         performanceClient?.addQueueMeasurement(PerformanceEvents.AuthorityFactoryCreateDiscoveredInstance, correlationId);
+        
+        const authorityUriFinal = Authority.transformCIAMAuthority(authorityUri);
```

Comment:
Can we do this in the authority constructor/instance methods instead?

Reply:
Guess we can. I chose this as most of the `authority` instance methods are getters and setters and having this code here made it more readable IMHO than in the other class. Let me wait for a few more reviews and make a call.

Reply:
I expect this to go away. The transition is temporary until STS supports the standard format without needing to append anything. So leaving as is for now.